### ISSUE TITLE
Show LSP diagnostic errors in `Project Errors`

### ISSUE THREAD

**Issue by angelozerr** [redhat-developer/lsp4ij#443]
Some language servers (ex: our Qute language servers) can report errors for closed document. It would be nice if we could see those erros in the project errors:

![image](https://github.com/user-attachments/assets/f2b368af-1633-41c2-8d91-74b79a5c4abf)

vcode provides this feature and can show all errors in the error view and for Qute for instance you can see all errors of all Qute templates (closed and opened):

![image](https://github.com/user-attachments/assets/30a5b355-0264-4b8c-aa88-9f79cec0be2a)

@ia3andy is it a feature that you would like to have?

I think this feature should be customized per language server definition and we should provide an API to know if LSP diagnostics must be displayed in `Project Errors` or not.

**Comment by acamargo** [comment]
Hello @angelozerr , thanks for raising this issue (I was about to create one myself).

Are there any plans to implement this?

A project-wide diagnostics overview would be very beneficial for improving developer awareness.

**Comment by angelozerr** [comment]
My short plan is to release LSP4IJ 0.12.0 but before we need to fix an annoying freeze issue https://github.com/redhat-developer/lsp4ij/discussions/899#discussioncomment-12637482

After the release I would like to implement this feature because for our Qute support we will need it //cc @ia3andy to show all errors of Qute template of the project and not only error from opened Qute template (in vscode we have this feature).

Have you some idea how to implement that?

**Comment by ericdallo** [comment]
@angelozerr makes sense, project aware errors is a must have for any LSP client IMO.
I don't know if lsp4ij register LSP diagnostics using the extension Annotator, because on clojure-lsp before lsp4ij integration, we used to have this for free using that extension

**Comment by angelozerr** [comment]
LSP4IJ uses an external annotator which register Annotation https://github.com/redhat-developer/lsp4ij/blob/ea0f108aa1677e844b405cb3d9d962e1535f07d8/src/main/java/com/redhat/devtools/lsp4ij/features/diagnostics/LSPDiagnosticAnnotator.java#L103

Have you some link how you managed that on clojure lsp?

**Comment by ericdallo** [comment]
@angelozerr [here](https://github.com/clojure-lsp/clojure-lsp-intellij/blob/1fb14a06decf4759fdc01b3766308a228b78baeb/src/main/clojure/com/github/clojure_lsp/intellij/extension/annotator.clj) is how we used to do, and [here](https://github.com/clojure-lsp/clojure-lsp-intellij/blob/1fb14a06decf4759fdc01b3766308a228b78baeb/src/main/resources/META-INF/plugin.xml#L30)'s the definition in plugin.xml

**Comment by angelozerr** [comment]
That's super strange, because I have the impression that LSP4IJ does the same thing than you?

**Comment by angelozerr** [comment]
@ericdallo I have created a PR at https://github.com/redhat-developer/lsp4ij/pull/936 

It doesn't fill the Project Errors but if you click on Inspect Code hyperlink it show the errors of all opened files.

I need to support now the closed file if language server can report diagnostics for closed files (like we have managed with our Qute LS)

**Comment by ericdallo** [comment]
@angelozerr thanks!
clojure-lsp server and multiple servers publish diagnostics of project files even if not opened, that's how it works for years, so it should be possible.

**Comment by angelozerr** [comment]
> @angelozerr thanks, although not very helpful for clojure.

why? because my PR doesn't support closed files? If it that,my goal is to support it because we need that for our Qute LS.

**Comment by ericdallo** [comment]
Yes, the closed files support is really crucial, thanks!

**Comment by angelozerr** [comment]
I have implemented closed document diagnostics, Iam testing with our Qute LS.

I have tried to consume lsp clojure but it doesn't publish diagnostics for closed documents?

**Comment by ericdallo** [comment]
It does @angelozerr if yu have any file with diagnostics, like a unresolved-symbol for example:
```clojure
(ns some-ns)

fooo ;; unresolved symbol here
```
